src/test/fuzz/socks5.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <netbase.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+bool Socks5(const std::string& strDest, int port, const ProxyCredentials* auth, const Sock& socket);
+
+namespace {
+int DEFAULT_SOCKS5_RECV_TIMEOUT;
+};
+
+extern int SOCKS5_RECV_TIMEOUT;
+void initialize_socks5()
+{
+    static const std::unique_ptr<const BasicTestingSetup> testing_setup = MakeFuzzingContext<>();
+    DEFAULT_SOCKS5_RECV_TIMEOUT = SOCKS5_RECV_TIMEOUT;
+}
+
+FUZZ_TARGET_INIT(socks5, initialize_socks5)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    struct ProxyCredentials proxy_credentials;
+    proxy_credentials.username = fuzzed_data_provider.ConsumeRandomLengthString(512);
+    proxy_credentials.password = fuzzed_data_provider.ConsumeRandomLengthString(512);
+    InterruptSocks5(fuzzed_data_provider.ConsumeBool());
+    // Set FUZZED_SOCKET_FAKE_LATENCY=1 to exercise recv timeout code paths. This
+    // will slow down fuzzing.
+    SOCKS5_RECV_TIMEOUT = (fuzzed_data_provider.ConsumeBool() && std::getenv("FUZZED_SOCKET_FAKE_LATENCY") != nullptr) ? 1 : DEFAULT_SOCKS5_RECV_TIMEOUT;
+    FuzzedSock fuzzed_sock = ConsumeSock(fuzzed_data_provider);
+    // This Socks5(...) fuzzing harness would have caught CVE-2017-18350 within
+    // a few seconds of fuzzing.
+    (void)Socks5(fuzzed_data_provider.ConsumeRandomLengthString(512), fuzzed_data_provider.ConsumeIntegral<int>(), fuzzed_data_provider.ConsumeBool() ? &proxy_credentials : nullptr, fuzzed_sock);
```

Comment:
nit: Would be nice to break long lines after each `,` or `?`, or `:`

Reply:
Fixed!

Reply:
> Would be nice to break long lines ...

Opened https://github.com/bitcoin/bitcoin/pull/21223 style: make clang-format break long lines